### PR TITLE
Update deployment for Kubernetes 1.31

### PR DESCRIPTION
**Pull request by bart0sh**
- Modified ClusterRole to include only API objects that the driver can access 
- Added ValidatingAdmissionPolicy for resourceslices

Fixes https://github.com/kubernetes-sigs/dra-example-driver/issues/48

### CODE REVIEW

**Comment by klueska** [comment]
Can we add the adminssion policy stuff as a separate PR where we also add support for requesting resources using AdminAccess?

**Comment by bart0sh** [comment]
@klueska done. Any ideas on where to add support for requesting resources using AdminAccess?

File: deployments/helm/dra-example-driver/templates/validatingadmissionpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
```
Comment (klueska) [inline deployments/helm/dra-example-driver/templates/validatingadmissionpolicy.yaml:2]: I'm actually not that familiar with `ValidatingAdmissionPolicy`. Can you explain to me in english what this file (in conjunction with [validatingadmissionpolicybinding.yaml](https://github.com/kubernetes-sigs/dra-example-driver/pull/66/files#diff-ae0655a5f1431437c30dc2197f4993c35a566fff2838db7a59992eebcd9a1d62)) is meant to do?

Comment (bart0sh) [reply deployments/helm/dra-example-driver/templates/validatingadmissionpolicy.yaml:2]: I understand it as a set of rules to restrict access from the client (DRA plugin in this case) to the API server. I used [test driver admin-access policy](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/dra/test-driver/deploy/example/admin-access-policy.yaml) as an example.

Comment (klueska) [reply deployments/helm/dra-example-driver/templates/validatingadmissionpolicy.yaml:2]: My question is more around what this gives us beyond what we get from the ClusterRole already? At a quick glance it looks as if it also limits each instance of a driver daemonset to only modify the resourceslices on the node it is running on. Is that correct? Is there more to it than that?

Comment (bart0sh) [reply deployments/helm/dra-example-driver/templates/validatingadmissionpolicy.yaml:2]: It also sets a list of acceptable API versions, `v1alpha3` in this particular case.

Comment (pohly) [reply deployments/helm/dra-example-driver/templates/validatingadmissionpolicy.yaml:2]: This is  just a selector for what the policy applies to. Once we have v1beta1 and a client uses it, this policy would be ignored.

**Review by klueska — APPROVED** [review]
/approve
/lgtm

**Pull request merged**